webapp/src/Service/Compare/ScoreboardCompareService.php

```diff
@@ -0,0 +1,134 @@
+                $this->addMessage(MessageType::ERROR, sprintf('Row %d: total time does not match', $index), (string)$row->score->totalTime, (string)$object2->rows[$index]->score->totalTime);
+            }
+
+            foreach ($row->problems as $problem) {
+                /** @var Problem|null $problemForSecond */
+                $problemForSecond = null;
+
+                foreach ($object2->rows[$index]->problems as $problem2) {
+                    // PC^2 uses different problem ID's. For now also match on `Id = {problemId}-{digits}`
+                    if ($problem->problemId === $problem2->problemId) {
+                        $problemForSecond = $problem2;
+                        break;
+                    } elseif (preg_match('/^Id = ' . preg_quote($problem->problemId, '/') . '-+\d+$/', $problem2->problemId)) {
+                        $problemForSecond = $problem2;
+                        break;
+                    }
+                }
+
+                if ($problemForSecond === null && $problem->solved) {
+                    $this->addMessage(MessageType::ERROR, sprintf('Row %d: Problem %s solved in first file, but not found in second file', $index, $problem->problemId));
+                } elseif ($problemForSecond !== null && $problem->solved !== $problemForSecond->solved) {
+                    $this->addMessage(MessageType::ERROR, sprintf('Row %d: Problem %s solved does not match', $index, $problem->problemId), (string)$problem->solved, (string)$problemForSecond->solved);
+                }
+
+                if ($problemForSecond) {
+                    if ($problem->numJudged !== $problemForSecond->numJudged) {
+                        $this->addMessage(MessageType::ERROR, sprintf('Row %d: Problem %s num judged does not match', $index, $problem->problemId), (string)$problem->numJudged, (string)$problemForSecond->numJudged);
+                    }
+
+                    if ($problem->numPending !== $problemForSecond->numPending) {
+                        $this->addMessage(MessageType::ERROR, sprintf('Row %d: Problem %s num pending does not match', $index, $problem->problemId), (string)$problem->numPending, (string)$problemForSecond->numPending);
+                    }
+
+                    if ($problem->time !== $problemForSecond->time) {
+                        // This is an info message for now, since PC^2 doesn't expose time info
+                        $this->addMessage(MessageType::INFO, sprintf('Row %d: Problem %s time does not match', $index, $problem->problemId), (string)$problem->time, (string)$problemForSecond->time);
+                    }
+                }
+            }
+
+            foreach ($object2->rows[$index]->problems as $problem2) {
+                $problemForFirst = null;
+
+                foreach ($row->problems as $problem) {
+                    // PC^2 uses different problem ID's. For now also match on `Id = {problemId}-{digits}`
```

Comment:
Are they correct according to the spec? If not we shouldn't do this as we decided to not have `PC2` specific fixes in the main codebase but have them in a separate branch.

Reply:
They are not. But we will not be able to automate checking if we see this as error since it will produce a non 0 exit code